src/translator/cache.h

```diff
@@ -0,0 +1,207 @@
+#ifndef SRC_BERGAMOT_CACHE_H_
+#define SRC_BERGAMOT_CACHE_H_
+
+#include <list>
+#include <mutex>
+#include <unordered_map>
+#include <utility>
+
+#ifndef WASM_COMPATIBLE_SOURCE
+#include "3rd_party/L4/inc/L4/LocalMemory/HashTableService.h"
+#endif
+
+#include "common/hash.h"
+#include "processed_request_sentence.h"
+#include "translator/beam_search.h"
+
+namespace marian {
+namespace bergamot {
+
+namespace cache_util {
+
+template <class HashWordType>
+struct HashWords {
+  HashWordType operator()(const Words &words) const {
+    size_t seed = 42;
+    for (auto &word : words) {
+      HashWordType hashWord = static_cast<HashWordType>(word.toWordIndex());
+      util::hash_combine<HashWordType>(seed, hashWord);
+    }
+    return seed;
+  }
+};
+
+}  // namespace cache_util
+
+struct CacheConfig {
```

Comment:
Could i have a small sweep of the bucket hyperparameter? Could you try a few values? I suspect lower bucket number might mean lower memory overhead, higher bucket number will mean less locking